build: switch to conda dssp installation

gcroci2 opened the pull request:
With this PR, `dssp` package is installed through conda instead of source or apt-get. This is a usage improvement for the users and is a step forward for deploying deeprank2 on conda (see issue #559). In particular:
- now the appropriate conda channels (pointing to the right packages' releases) are specified in the action file
- [dssp (version 4.2.2.1)](https://anaconda.org/sbl/dssp) is installed via conda
- this solves #542, by placing an init file into the tests folder. This is solved by #554 as well since Ruff automatically added the init file to the tests folder (and each subfolder, even if is not necessary for coveralls to detect all the files). I also added a check in the coveralls workflow file to make the action fail if the coverage drops below 80%.
- just to be sure since we do not have an action for testing the dockerfile, be sure to run that procedure again

DaniBodor commented on .github/actions/install-python-and-package/action.yml:
```diff
@@ -27,8 +27,10 @@ runs:
-    - run: conda --version
+        conda-channels: pytorch, pyg, bioconda, defaults, sbl, conda-forge
+    - run: |
+        conda --version
+        conda env list
```
what's the point of this? Just to for inspection in case it fails?

gcroci2 replied:
yes :)

DaniBodor reviewed (approved):
Looks good! I tested all 3 installation procedures (docker, env file, manual) and they all work!

I added 2 commits to this branch. 
1. The first one is some automatic formatting updates. I think maybe you don't have the prettier extension installed, so then vs code cannot automatically format the files that use that (most non-python files).
2. Suggestions to the README. 
I thought it'd be easier to just put everything in a single commit, rather than add a bunch of code suggestions. Feel free to drop either if you don't like it or to adjust as you prefer.


Finally, the newest version of pandas is giving a warning:

> /tmp/ipykernel_32/1088082951.py:2: DeprecationWarning: 
> Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
> (to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
> but was not found to be installed on your system.
> If this would cause problems for you,
> please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
>         
>   import pandas as pd

Maybe we should add Pyarrow to the deeprank dependencies to make it future proof (and avoid the warning).

DaniBodor commented on docs/installation.md:
would be nice to add a table of contents here as well

gcroci2 commented on README.md:
```diff
@@ -34,76 +34,126 @@ DeepRank2 extensive documentation can be found [here](https://deeprank2.rtfd.io/
+  - [Datasets](#datasets)
+    - [GraphDataset](#graphdataset)
+    - [GridDataset](#griddataset)
+  - [Training](#training)
```
When I save the file in vscode, it does it automatically for me. I think it's because I have [this extension](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)

gcroci2 commented on .github/actions/install-python-and-package/action.yml:
```diff
@@ -59,26 +61,7 @@ runs:
-        mkdir build
-        cmake -S . -B build -DCMAKE_BUILD_TYPE=Release
-        cmake --build build
-        cmake --install build
+        conda install -c sbl dssp>=4.2.2.1
```
This is for `if runner.os == 'macOS'`, while the one in line 48 is for linux. We're not running the CI on mac yet, but I keep it updated since it's in my todo list

gcroci2 commented on README.md:
```diff
@@ -34,54 +34,100 @@ DeepRank2 extensive documentation can be found [here](https://deeprank2.rtfd.io/
-## Installation
+## Installations
 
 The package officially supports ubuntu-latest OS only, whose functioning is widely tested through the continuous integration workflows.
```
I am keeping your commit so these comments are outdated (all included in your commit)

gcroci2 commented on README.md:
```diff
@@ -34,54 +34,100 @@ DeepRank2 extensive documentation can be found [here](https://deeprank2.rtfd.io/
+    * [Here](https://ssbio.readthedocs.io/en/latest/instructions/msms.html) for MacOS with M1 chip users.
+*  [PyTorch](https://pytorch.org/get-started/locally/)
+    * We support torch's CPU library as well as CUDA.
+    * Currently, the package is tested using [PyTorch 2.0.1](https://pytorch.org/get-started/previous-versions/#v201).   
```
Updated thanks :)

gcroci2 commented on .github/actions/install-python-and-package/action.yml:
Totally! I've already listed it as a TODO in #559. I tried already here to do it, but for some reason, I was getting inconsistent errors and I decided to do it separately from this PR.

gcroci2 commented:
> Looks good! I tested all 3 installation procedures (docker, env file, manual) and they all work!
> 
> I added 2 commits to this branch.
> 
> 1. The first one is some automatic formatting updates. I think maybe you don't have the prettier extension installed, so then vs code cannot automatically format the files that use that (most non-python files).

Oh right, I installed it now :)

> 2. Suggestions to the README.
>    I thought it'd be easier to just put everything in a single commit, rather than add a bunch of code suggestions. Feel free to drop either if you don't like it or to adjust as you prefer.

Very nice edits, thanks! I'll leave the commit, I only corrected two typos and edited a couple of subtitles. 

> Note that the python 3.11 issue isn't (stably) fixed after all, so please do not close that issue with this PR.

I removed it from my initial comment indeed. What if I remove Python 3.11 from the CI for now? In a few weeks we can try to pick up the issue again, check it better, and see if anything has changed. But at least we will publish a "real" stable version that works on Python 3.10. I have added a commit for this, let me know what you think. If you agree, I will also add a comment to issue #540. 

> Finally, the newest version of pandas is giving a warning. Maybe we should add Pyarrow to the deeprank dependencies to make it future proof (and avoid the warning).

I think they will add it themselves when installing pandas (as a pandas dependency in their config/toml file). The warning should be to gather feedback on this possibility shortly so that they know if adding this dependency will cause an issue for anyone. But I wouldn't worry about it, it will be installed automatically by pandas when the time comes.

DaniBodor commented on docs/installation.md:
```diff
@@ -1,49 +1,107 @@
-The `test` extra is optional, and can be used to install test-related dependencies useful during the development.
+The `test` extra is optional, and can be used to install test-related dependencies, useful during development.
+
+#### Testing DeepRank2 installation
```
```suggestion
### Testing DeepRank2 installation
```

DaniBodor commented on docs/installation.md:
```diff
@@ -1,49 +1,107 @@
+  - [Containerized Installation](#containerized-installation)
+  - [Local/remote installation](#localremote-installation)
+      - [YML file installation](#yml-file-installation)
+    - [Manual installation](#manual-installation)
+      - [Testing DeepRank2 installation](#testing-deeprank2-installation)
```
```suggestion
    - [Manual installation](#manual-installation)
    - [Testing DeepRank2 installation](#testing-deeprank2-installation)
```

DaniBodor reviewed (approved):
I noticed an indent level error in the installation.md file (which I probably created when copy/pasting from the README).

DaniBodor commented:
> > Looks good! I tested all 3 installation procedures (docker, env file, manual) and they all work!
> > I added 2 commits to this branch.
> > 
> > 1. The first one is some automatic formatting updates. I think maybe you don't have the prettier extension installed, so then vs code cannot automatically format the files that use that (most non-python files).
> 
> Oh right, I installed it now :)
> 
> > 2. Suggestions to the README.
> >    I thought it'd be easier to just put everything in a single commit, rather than add a bunch of code suggestions. Feel free to drop either if you don't like it or to adjust as you prefer.
> 
> Very nice edits, thanks! I'll leave the commit, I only corrected two typos and edited a couple of subtitles.
> 
> > Note that the python 3.11 issue isn't (stably) fixed after all, so please do not close that issue with this PR.
> 
> I removed it from my initial comment indeed. What if I remove Python 3.11 from the CI for now? In a few weeks we can try to pick up the issue again, check it better, and see if anything has changed. But at least we will publish a "real" stable version that works on Python 3.10. I have added a commit for this, let me know what you think. If you agree, I will also add a comment to issue #540. @DaniBodor
> 
> > Finally, the newest version of pandas is giving a warning. Maybe we should add Pyarrow to the deeprank dependencies to make it future proof (and avoid the warning).
> 
> I think they will add it themselves when installing pandas (as a pandas dependency in their config/toml file). The warning should be to gather feedback on this possibility shortly so that they know if adding this dependency will cause an issue for anyone. But I wouldn't worry about it, it will be installed automatically by pandas when the time comes.

Perfect. I agree with all of this.
